code/modules/supply/supply_packs/pack_abstract.dm

```diff
@@ -0,0 +1,93 @@
+	/// The shuttle that this pack corresponds to ordering.
+	/// If given on init, will fill out the pack automagically.
+	var/datum/map_template/shuttle/template
+	/// The base speed multiplier to apply to the shuttle. 2 means it arrives twice as fast.
+	var/speed_factor = 1
+
+
+/datum/supply_packs/abstract/shuttle/New(datum/map_template/shuttle/_template)
+	. = ..()
+	if(_template)
+		template = _template
+	if(template)
+		if(ispath(template))
+			template = new template()
+		name = template.name
+		manifest = template.description
+		ui_manifest = list(manifest)
+
+/datum/supply_packs/abstract/shuttle/Destroy(force, ...)
+	. = ..()
+	qdel(template)
+
+
+/datum/supply_packs/abstract/shuttle/can_order()
+	// check here so people don't spend their money on it
+	return (SSshuttle.emergency?.mode == SHUTTLE_IDLE)
+
+/datum/supply_packs/abstract/shuttle/on_order_confirm(datum/supply_order/order)
+	. = ..()
+	if(isnull(template))
+		CRASH("Shuttle pack [src] has no map template to load!")
+	GLOB.major_announcement.Announce("[order.orderedby] has purchased [name] for [cost] credit\s as the emergency shuttle for the shift.", "Shuttle Purchase Receipt")
+	var/orderer = order.orderedby
+	log_and_message_admins("Shuttle pack [src] costing [cost] has been ordered and paid for by [orderer] ([order.orderedbyRank])")
+	for(var/datum/supply_packs/abstract/shuttle/shuttle_pack in SSeconomy.supply_packs)
+		shuttle_pack.special_enabled = FALSE
+	SSshuttle.custom_shuttle_ordered = TRUE
+	// var/datum/map_template
+	// TODO prevent this from happening while someone else is modifying it, or while it's not at centcomm
+	// probably should do this in a way that prevents people from buying it
+	var/obj/docking_port/mobile/shuttle = SSshuttle.load_template(template)
+	shuttle.shuttle_speed_factor = speed_factor
+	SSshuttle.replace_shuttle(shuttle)
+
+/datum/supply_packs/abstract/admin_notify/donations
+	name = "Donation to Lonely Corgi Foundation"
+	manifest = "1% of every donation goes towards supporting corgis in need."
+	cost = 500
+
+
+/datum/supply_packs/abstract/shuttle/bar
+	template = /datum/map_template/shuttle/emergency/bar
+
+/datum/supply_packs/abstract/shuttle/dept
+	template = /datum/map_template/shuttle/emergency/dept
+
+/datum/supply_packs/abstract/shuttle/military
+	template = /datum/map_template/shuttle/emergency/military
+
+/datum/supply_packs/abstract/shuttle/clown
+	speed_factor = 5
+	contraband = TRUE
+	template = /datum/map_template/shuttle/emergency/clown
+
+/datum/supply_packs/abstract/shuttle/cramped
+	template = /datum/map_template/shuttle/emergency/cramped
+
+/datum/supply_packs/abstract/shuttle/narnar
+	contraband = TRUE
+	template = /datum/map_template/shuttle/emergency/narnar
+
+/datum/supply_packs/abstract/shuttle/old
+	template = /datum/map_template/shuttle/emergency/old
+
+/datum/supply_packs/abstract/shuttle/jungle
+	template = /datum/map_template/shuttle/emergency/jungle
```

Comment:
minimum $2500 credits. 
-Anything with potentially hazardous stuff on-board or little room should be upwards $3500

Reply:
Feel free to take another look. I added some base prices and made the clown shuttle cmag-only